DemoApp/App.js

```diff
@@ -56,11 +64,52 @@ const App: () => React$Node = () => {
       viewportHeight: 456
     });
 
-  Tracker.trackScreenViewEvent('Name', null, null, null, null, null, null, []);
-  const onPressSendEvent = () => {
-    Tracker.trackSelfDescribingEvent({'schema': 'iglu:com.acme/event/jsonschema/1-0-0', 'data': {'message': 'hello world'}}, []);
-    Tracker.trackStructuredEvent('category', 'action', 'label', 'property', 50.00, []);
-    Tracker.trackPageViewEvent('acme.com', 'some title', 'refr.com', []);
+  initPromise.then(() => Tracker.trackScreenViewEvent({screenName: 'firstScreenView'}));
+
+  const onPresstrackStructuredEvent = () => {
+    Tracker.trackStructuredEvent({category: 'SeTest', action: 'allPopulated', label: 'valueIsFloat', property: 'property', value: 50.00});
```

Comment:
Should this be:

```
const onPresstrackStructuredEvent = () =>
  initPromise.then(() =>
    Tracker.trackStructuredEvent(..... // etc
```

Depending on how paranoid you want to be about the tracker not being initialized yet.

Reply:
So, yes - it's a bit wonky the way I've done it, but I did it this way on purpose.

What I was aiming for was a test case which uses the promise, and a test case which doesn't, under the assumption that some won't use this structure.

Thinking about that now though, having stepped back 'out of the weeds' - I probably shouldn't design for a non-deterministic way of using the tracker. I should probably both document and test things to use the preferable pattern, and leave those who don't follow it to their own devices...